e2e/cypress/integration/enterprise/saml/saml_guest_member_spec.js

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @enterprise @saml
+
+import * as TIMEOUTS from '../../../fixtures/timeouts';
+import {getRandomId} from '../../../utils';
+
+// assumes that E20 license is uploaded
+// Update config.mk to make sure docker images for openldap and keycloak
+//  - assumes openldap docker available on config default http://localhost:389
+//  - assumes keycloak docker - uses api to update
+// assumes the CYPRESS_* variables are set (CYPRESS_keycloakBaseUrl / CYPRESS_keycloakAppName)
+// requires {"chromeWebSecurity": false}
+// copy ./mattermost-server/build/docker/keycloak/keycloak.crt -> ./mattermost-webapp/e2e/cypress/fixtures/keycloak.crt
+describe('SAML Guest', () => {
+    const loginButtonText = 'SAML';
+
+    const guestUser = {
+        username: 'guest.test',
+        password: 'Password1',
+        email: 'guest.test@mmtest.com',
+        firstname: 'Guest',
+        lastname: 'OneSaml',
+        keycloakId: '',
+    };
+    const userFilter = `username=${guestUser.username}`;
+    const keycloakBaseUrl = Cypress.env('keycloakBaseUrl') || 'http://localhost:8484';
+    const keycloakAppName = Cypress.env('keycloakAppName') || 'mattermost';
+    const idpUrl = `${keycloakBaseUrl}/auth/realms/${keycloakAppName}/protocol/saml`;
+    const idpDescriptorUrl = `${keycloakBaseUrl}/auth/realms/${keycloakAppName}`;
+
+    const newConfig = {
```

Comment:
There was a recent change in sidebar config. In Cypress, it's currently set to false by default. With that, please add the following since you're using `AddChannelDropdown_dropdownButton` to validate guest or regular user's membership.
```
ServiceSettings: {
    EnableLegacySidebar: false,
},
```

Reply:
The default actually changed while I was writing it. I had to change the test... :) Done.